src/components/VenueBody/DatePicker/useDatePicker.js

```diff
@@ -0,0 +1,91 @@
+import { useEffect, useState } from "react";
+
+export const useDatePicker = () => {
+  const [whichCalendarIsActive, setWhichCalendarIsActive] = useState("start");
+  const [startDate, setStartDate] = useState(null);
+  const [endDate, setEndDate] = useState(null);
+  const [isChecked, setIsChecked] = useState(false);
+  const [isCalendarError, setIsCalendarError] = useState(false);
+  const [daysBetween, setDaysBetween] = useState(0);
+
+  useEffect(() => {
+    const calculateDaysBetween = () => {
+      if (!startDate || !endDate) {
+        setDaysBetween(0);
+      } else if (startDate === endDate) {
+        setDaysBetween(1);
+      } else {
+        const diff = Math.ceil(
+          (new Date(endDate) - new Date(startDate)) / (1000 * 60 * 60 * 24),
+        );
+        setDaysBetween(diff);
+      }
+    };
+    calculateDaysBetween();
+  }, [startDate, endDate]);
```

Comment:
This function is not async, no point of creating it and then calling it. Just put the logic in well-named effect function

Reply:
done